src/frontend/src/pages/DesignReviewSummaryModal.tsx

```diff
@@ -0,0 +1,285 @@
+import DescriptionIcon from '@mui/icons-material/Description';
+import VideocamIcon from '@mui/icons-material/Videocam';
+import EditIcon from '@mui/icons-material/Edit';
+import Checkbox from '@mui/material/Checkbox';
+import SaveIcon from '@mui/icons-material/Save';
+import { ChangeEvent, useState } from 'react';
+import { DesignReviewDelayModal } from '../components/DesignReviewDelayModal';
+import { DesignReviewPill } from '../components/DesignReviewPill';
+import { DesignReviewMemberPill } from '../components/DesignReviewMemberPill';
+import NERFailButton from '../components/NERFailButton';
+import NERSuccessButton from '../components/NERSuccessButton';
+import { NERButton } from '../components/NERButton';
+import StageGateWorkPackageModalContainer from './WorkPackageDetailPage/StageGateWorkPackageModalContainer/StageGateWorkPackageModalContainer';
+import { availabilityStartTimePipe, meetingDatePipe } from '../utils/pipes';
+import { useCurrentUser } from '../hooks/users.hooks';
+import { useToast } from '../hooks/toasts.hooks';
+import { getTeamTypeIcon } from '../utils/design-review.utils';
+
+interface DRCSummaryModalProps {
+  open: boolean;
+  onHide: () => void;
+  designReview: DesignReview;
+}
+
+const DRCSummaryModal: React.FC<DRCSummaryModalProps> = ({ open, onHide, designReview }) => {
+  const [requiredMembers, setRequiredMembers] = useState<User[]>(designReview.requiredMembers);
+  const [optionalMembers, setOptionalMembers] = useState<User[]>(designReview.optionalMembers);
+  const [checked, setChecked] = useState<boolean>(false);
+  const [showStageGateModal, setShowStageGateModal] = useState<boolean>(false);
+  const [showDelayModal, setShowDelayModal] = useState<boolean>(false);
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setChecked(event.target.checked);
+  };
+  const [editing, setEditing] = useState<boolean>(false);
+  const [locationState, setLocationState] = useState<string>(() => designReview.location || '');
+  const [documentState, setDocumentState] = useState<string>(() => designReview.docTemplateLink || '');
+  const [zoomState, setZoomState] = useState<string>(() => designReview.zoomLink || '');
+  const currentUser = useCurrentUser();
+  const toast = useToast();
+  const handleRemoveRequiredMember = (user: User) => {
+    if (currentUser.userId === designReview.userCreated.userId) {
+      setRequiredMembers(requiredMembers.filter((member) => member.userId !== user.userId));
+    } else {
+      toast.error('Only the creator of the Design Review can edit attendees');
+    }
+  };
+
+  const handleRemoveOptionalMember = (user: User) => {
+    if (currentUser.userId === designReview.userCreated.userId) {
+      setOptionalMembers(optionalMembers.filter((member) => member.userId !== user.userId));
+    } else {
+      toast.error('Only the creator of the Design Review can edit attendees');
+    }
+  };
+
+  const ModalButtons: React.FC = () => (
```

Comment:
I would move all these components you defined into their own separate files where u can because this file is kind of heinous rn